consent/strategy_default.go

```diff
@@ -220,12 +223,6 @@ func (s *DefaultStrategy) forwardAuthenticationRequest(
 		skip = true
 	}
 
-	// Let's validate that prompt is actually not "none" if we can't skip authentication
-	prompt := stringsx.Splitx(ar.GetRequestForm().Get("prompt"), " ")
-	if stringslice.Has(prompt, "none") && !skip {
-		return errorsx.WithStack(fosite.ErrLoginRequired.WithHint(`Prompt 'none' was requested, but no existing login session was found.`))
-	}
-
```

Comment:
In my understanding, this contract is broken now with the change. I don't think that subject == "" is equal to `.authenticationSession(ctx, w, r)` returning `ErrNoAuthenticationSessionFound`.